arangod/Aql/IndexExecutor.h

```diff
@@ -117,6 +118,10 @@ class IndexExecutorInfos {
     return _outNonMaterializedIndRegs;
   }
 
+  bool isOneIndexCondition() const noexcept {
+    return _oneIndexCondition;
+  }
```

Comment:
As far as I can see, the `_oneIndexCondition` boolean is used only as a performance optimization if there is only a single lookup condition (in contrast to having multiple sub-conditions combined with logical OR).
What we will gain from it is to save a call to `getMember(...)` on the conditions root node.
If this is the only thing it does but is otherwise not required, I think it may be worthwhile to drop the boolean because it creates more complicated code and special handling in a few files.
At least from the logical point of view, having only a single condition should be a subset of having n ORed conditions.